[weakaura] missing field `username`

**Describe the bug**
Stuck at parsing weakauras

**To Reproduce**
Steps to reproduce the behavior:
1. Go to "My Weakauras"
2. Click on dropdown on right side
3. Select my username
4. Get stuck

**Expected behavior**
Show weakauras

**Screenshots**
![image](https://user-images.githubusercontent.com/17293360/102757205-41d86c00-4371-11eb-9b37-27c65c9f3cae.png)


**Software involved**
Windows 10 20H2
WoW Classic
Ajour 0.6.0

**Log Output**
```
09:37:58.456 [ajour::gui::update][DEBUG] Message::ParsedAddons(Classic, 63 addons)
09:37:58.794 [ajour_core::catalog][DEBUG] Successfully fetched and parsed https://github.com/casperstorm/ajour-catalog/releases/latest/download/wowi.json
09:37:59.195 [ajour_core::catalog][DEBUG] Successfully fetched and parsed https://github.com/casperstorm/ajour-catalog/releases/latest/download/curse.json
09:37:59.242 [ajour::gui::update][DEBUG] Message::CatalogDownloaded(15671 addons in catalog)
09:38:03.107 [ajour::gui::update][DEBUG] Interaction::ModeSelected(MyWeakAuras(Classic))
09:38:08.333 [ajour::gui::update][DEBUG] Message::WeakAurasAccountSelected(**HIDDEN**)
09:38:08.643 [ajour][ERROR] Failed to parse WeakAuras
09:38:08.643 [ajour][ERROR] caused by: missing field `username` at line 1 column 1868
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,10 @@ and `Removed`.
 
 ## [Unreleased]
 
+### Fix
+
+- Now correctly parses WeakAuars uploaded to Wago as a guest.
+
 ## [0.6.0] - 2020-12-20
 
 ### Added

--- a/crates/weak_auras/src/lib.rs
+++ b/crates/weak_auras/src/lib.rs
@@ -183,7 +183,7 @@ impl AuraUpdate {
 
         writeln!(&mut slug, "    [\"{}\"] = {{", self.slug)?;
         writeln!(&mut slug, "      name = [=[{}]=],", self.aura.name)?;
-        writeln!(&mut slug, "      author = [=[{}]=],", self.aura.author().to_owned())?;
+        writeln!(&mut slug, "      author = [=[{}]=],", self.aura.author())?;
         writeln!(&mut slug, "      encoded = [=[{}]=],", self.encoded_update)?;
         writeln!(&mut slug, "      wagoVersion = [=[{}]=],", self.aura.version)?;
         writeln!(&mut slug, "      wagoSemver = [=[{}]=],", self.aura.version_string)?;

--- a/crates/weak_auras/src/lib.rs
+++ b/crates/weak_auras/src/lib.rs
@@ -183,7 +183,7 @@ impl AuraUpdate {
 
         writeln!(&mut slug, "    [\"{}\"] = {{", self.slug)?;
         writeln!(&mut slug, "      name = [=[{}]=],", self.aura.name)?;
-        writeln!(&mut slug, "      author = [=[{}]=],", self.aura.username)?;
+        writeln!(&mut slug, "      author = [=[{}]=],", self.aura.author().to_owned())?;
         writeln!(&mut slug, "      encoded = [=[{}]=],", self.encoded_update)?;
         writeln!(&mut slug, "      wagoVersion = [=[{}]=],", self.aura.version)?;
         writeln!(&mut slug, "      wagoSemver = [=[{}]=],", self.aura.version_string)?;
@@ -264,7 +264,7 @@ impl Display for AuraStatus {
 pub struct Aura {
     slug: String,
     name: String,
-    username: String,
+    username: Option<String>,
     version: u16,
     version_string: String,
     changelog: AuraChangelog,
@@ -329,7 +329,10 @@ impl Aura {
     }
 
     pub fn author(&self) -> &str {
-        &self.username
+        match &self.username {
+            Some(username) => username,
+            None => "Unknown",
+        }
     }
 
     fn parent_display(&self) -> Option<&AuraDisplay> {

